docs: add `run bundle` documentation (#4424)
operator-framework/operator-sdk: website/content/en/docs/olm-integration/quickstart-bundle.md

$ make bundle
/home/user/go/bin/controller-gen "crd:trivialVersions=true" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
operator-sdk generate kustomize manifests -q
kustomize build config/manifests | operator-sdk generate bundle -q --overwrite --version 0.0.1
INFO[0000] Building annotations.yaml
INFO[0000] Writing annotations.yaml in /home/user/go/src/github.com/test-org/memcached-operator/bundle/metadata
INFO[0000] Building Dockerfile
INFO[0000] Writing bundle.Dockerfile in /home/user/go/src/github.com/test-org/memcached-operator
operator-sdk bundle validate ./bundle
INFO[0000] Found annotations file                        bundle-dir=bundle container-tool=docker
INFO[0000] Could not find optional dependencies file     bundle-dir=bundle container-tool=docker
INFO[0000] All validation tests have completed successfully
```

> The above command will have created the bundle manifests' directories: a manifests directory

Reviewer: ```suggestion
The above command will have created the following bundle artifacts: a manifests directory
```

### Testing bundles

> In order to test our bundle in a cluster with [OLM installed](#enabling-olm).

Reviewer: Since this is a guide, the reader is expected to have installed OLM already after reading the `Enabling OLM` section.
```suggestion
```

> First, we need to build our bundle. To build a memcached-operator bundle, run:

Reviewer: To make it clear that these bundles are versioned alongside the operator
```suggestion
First, we need to build our bundle. To build a memcached-operator bundle for version v0.0.1, run:
```

```console

> make bundle-build BUNDLE_IMG=<some-registry>/memcached-operator-bundle:<tag>

Reviewer: ```suggestion
make bundle-build BUNDLE_IMG=<some-registry>/memcached-operator-bundle:v0.0.1
```

> docker push <some-registry>/memcached-operator-bundle:<tag>

Reviewer: ```suggestion
make docker-push IMG=<some-registry>/memcached-operator-bundle:v0.0.1
```

```

> With the bundle pushed, we can run it with [run bundle][cli-run-bundle].
> The `operator-sdk run bundle` will add your bundle to an emphemeral index
> in the cluster and deploy your operator with OLM.

Reviewer: ```suggestion
Now that the bundle image is present in a registry, [`operator-sdk run bundle`][cli-run-bundle]
can create a pod to serve that bundle to OLM via a [`Subscription`](https://olm.operatorframework.io/docs/tasks/install-operator-with-olm/#install-your-operator),
along with other OLM objects, ephemerally:
```

```console

> operator-sdk run bundle <some-registry>/memcached-operator-bundle:<tag>

Reviewer: ```suggestion
operator-sdk run bundle <some-registry>/memcached-operator-bundle:v0.0.1
```

### Testing bundles

> Before proceeding, make sure you've [Installed OLM](#enabling-olm) onto your
> cluster. First, we need to build our bundle. To build a memcached-operator

Reviewer: Since the prereq is important it should probably float
```suggestion
Before proceeding, make sure you've [Installed OLM](#enabling-olm) onto your cluster.

First, we need to build our bundle. To build a memcached-operator
```

bundle, run:

```console
make bundle-build BUNDLE_IMG=<some-registry>/memcached-operator-bundle:v0.0.1
make docker-push IMG=<some-registry>/memcached-operator-bundle:v0.0.1
```

Now that the bundle image is present in a registry, [`operator-sdk run bundle`][cli-run-bundle]
can create a pod to serve that bundle to OLM via a [`Subscription`][install-your-operator],
along with other OLM objects, ephemerally.

```console

> operator-sdk run bundle <some-registry>/memcached-operator-bundle:v0.0.1

Reviewer: ```suggestion
$ operator-sdk run bundle <some-registry>/memcached-operator-bundle:v0.0.1
```
